src/messages/messages.service.ts

```diff
@@ -19,6 +19,12 @@ export class MessagesService {
     return insertMessage;
   }
 
+  async createOfflineMessages(data, userId) {
+    data.map(async res => await this.messageRepository.insert({authorId: userId, message: res.message, createdAt: res.createdAt}));
```

Comment:
Do not insert rows in loop. That causes DB query insertion per each iteration.

Reply:
Fixed